pkg/profilecreator/profilecreator.go

```diff
@@ -254,8 +254,12 @@ func (ghwHandler GHWHandler) SortedTopology() (*topology.Info, error) {
 
 // GetReservedAndIsolatedCPUs returns Reserved and Isolated CPUs
 func (ghwHandler GHWHandler) GetReservedAndIsolatedCPUs(reservedCPUCount int, splitReservedCPUsAcrossNUMA bool) (string, string, error) {
-	if reservedCPUCount < 0 {
-		return "", "", fmt.Errorf("Specified eservered CPU count is negative, please specify it correctly")
+	cpuInfo, err := ghwHandler.CPU()
+	if err != nil {
+		return "", "", fmt.Errorf("Error obtaining CPU Info from GHW snapshot: %v", err)
+	}
+	if reservedCPUCount < 0 || reservedCPUCount > int(cpuInfo.TotalThreads) {
+		return "", "", fmt.Errorf("Specified reserved CPU count is invalid, please specify it correctly")
```

Comment:
this is very nicely worded but not helpful. Let's tell the user what PPC thinks is correct:
```
fmt.Errorf("Specified reserved CPU count is invalid, please specify it in the range [0,%d]", cpuInfo.TotalThreads)

Reply:
+1, will do!